### PR TITLE
Enhancements

### PR DESCRIPTION
**Pull request by tomossomot**
Issues [11](https://github.com/commerceblock/coordinator/issues/11), [14](https://github.com/commerceblock/coordinator/issues/14) and [31](https://github.com/commerceblock/coordinator/issues/31). Each issue more or less in its own commit for easy reviewing.

### CODE REVIEW

File: src/challenger.rs [hunk 1]
```diff
@@ -390,6 +388,7 @@ mod tests {
             time::Duration::from_millis(10),
             time::Duration::from_millis(10),
             3,
+            time::Duration::from_secs(1),
```
Comment (nkostoulas) [inline src/challenger.rs]: You can make it millis as well like the other timings to speed up tests.

**Comment by tomossomot** [comment]
Yeah lets leave it as is for now and we can move things around if testing-utils becomes more cluttered. 

Demo works as before. Challenges are eventually sent and verified but Its outputs are things such as: 
`400 Bad Request`
`body: bad-bid`
I'm assuming these are caused by the mock Service stuff. Any idea if this is expected behaviour or do you think it is worth looking in to?

File: examples/demo.rs [hunk 2]
```diff
@@ -46,21 +47,26 @@ fn main() {
         )
         .unwrap(),
     );
+    println!("client chain: {}",config.clientchain.host);
```
Comment (nkostoulas) [inline examples/demo.rs]: Remove this

File: src/clientchain.rs [hunk 3]
```diff
@@ -2,29 +2,29 @@
 //!
 //! Client chain interface and implementations
 
-use std::cell::RefCell;
 use std::collections::HashMap;
 
-use bitcoin_hashes::{hex::FromHex, sha256d, Hash};
+use bitcoin_hashes::{hex::FromHex, sha256d};
 use ocean_rpc::{json, RpcApi};
+use std::str;
```
Comment (nkostoulas) [inline src/clientchain.rs]: why is this needed?

File: src/checks.rs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+}
+/// Check for correct hash input string format
+pub fn check_hash_string(str: &String) -> bool {
+    if str.len() == 64 && str.chars().all(|x| x.is_ascii_hexdigit()) {
+        return true;
+    }
+    return false;
+}
+/// Check for correct host input string format
+pub fn check_host_string(str: &String) -> bool {
```
Comment (nkostoulas) [inline src/checks.rs]: This is not very useful. Some host addresses might not even have `:`. Probably best to remove it.

File: src/config.rs [hunk 5]
```diff
@@ -174,7 +173,12 @@ impl Config {
         // CO_CLIENTCHAIN__HOST=127.0.0.1:5555
         // CO_CLIENTCHAIN__GENESIS_HASH=706f6...
         if let Ok(v) = env::var("CO_API_HOST") {
-            let _ = conf_rs.set("api.host", v)?;
+            if check_host_string(&v) {
```
Comment (nkostoulas) [inline src/config.rs]: Env are set from the default.toml file, from implicit envs (e.g. CO_CHALLENG_DURATION) and from explicit envs. You are only checking for the explicit ones. I think you should check for all. But it will only require checking genesis hash and asset key.

Comment (nkostoulas) [reply src/config.rs]: Maybe there is a way by overriding the constructor not sure though.

File: src/storage.rs [hunk 6]
```diff
@@ -48,11 +47,11 @@ impl MongoStorage {
         let client = Client::with_uri(&uri)?;
 
         let db = client.db("coordinator");
-        if let Some(ref user) = storage_config.user {
-            if let Some(ref pass) = storage_config.pass {
-                db.auth(user, pass)?;
-            }
-        }
+        // if let Some(ref user) = storage_config.user {
```
[the remark this answers is not in the document]
Comment (nkostoulas) [reply src/storage.rs]: This is not resolved.

File: .rustfmt.toml [hunk 7]
```diff
@@ -1,5 +1,5 @@
 format_strings = true
 wrap_comments = true
-format_doc_comments = true
+format_code_in_doc_comments = true
```
[the remark this answers is not in the document]
Comment (tomossomot) [reply .rustfmt.toml]: It got renamed so `format_doc_comments` no longer works.
[rustfmt docs](https://rust-lang.github.io/rustfmt/#format_code_in_doc_comments)
[change](https://github.com/rust-lang/rustfmt/pull/3535)

[the remark this answers is not in the document]
Comment (nkostoulas) [reply .rustfmt.toml]: Oh okay

**Comment by nkostoulas** [comment]
Everything else looks good

**Comment by tomossomot** [comment]
cool, simplified!
